device/iot-device-client/src/main/java/com/microsoft/azure/sdk/iot/device/transport/IotHubTransport.java

```diff
@@ -30,47 +30,54 @@
 public class IotHubTransport implements IotHubListener
 {
     private static final int MAX_MESSAGES_TO_SEND_PER_THREAD = 10;
+
+    // For tracking the state of this layer in particular. If multiplexing, this value may be CONNECTED while a
+    // device specific state is DISCONNECTED_RETRYING. If this state is DISCONNECTED_RETRYING, then the multiplexed
+    // connection will be completely torn down and re-opened.
     private volatile IotHubConnectionStatus connectionStatus;
+
+    // for multiplexing. A particular device can be disconnected retrying while the tcp connection is fine and the other
+    // device sessions are open.
+    private Map<String, IotHubConnectionStatus> deviceConnectionStatuses = new HashMap<>();
```

Comment:
Maybe this is just me, but I cringe at the word "statuses". I much prefer "states". Thoughts?